subsys/bluetooth/host/l2cap.c

```diff
@@ -1193,7 +1190,15 @@ static void le_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 	rsp->result = BT_L2CAP_LE_SUCCESS;
 
 rsp:
-	l2cap_send(conn, BT_L2CAP_CID_LE_SIG, buf);
+	if (bt_l2cap_send(conn, BT_L2CAP_CID_LE_SIG, buf)) {
```

Comment:
There seems to be inconsistency with using `result` vs `rsp->result` in this function. In some cases the value of `result` gets ignored and some other value gets encoded into `rsp->result`. Since you're updating the code flow, I'd suggest to fix this so that there's a single place that encodes `rsp->result`, right after the `rsp:` label:
```c
	rsp->result = sys_cpu_to_le16(result);
```

So that means that all the places with `goto rsp;` should set `result` instead of `rsp->result`.